conditions/certificaterequest.go

```diff
@@ -17,30 +17,61 @@ limitations under the License.
 package conditions
 
 import (
-	cmutil "github.com/cert-manager/cert-manager/pkg/api/util"
 	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
 	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/clock"
 )
 
 // Update the status with the provided condition details & return
 // the added condition.
-// NOTE: this code is just a workaround for cmutil only accepting the certificaterequest object
 func SetCertificateRequestStatusCondition(
```

Comment:
comment: It's a shame to have this code here - functions which act on cert-manager resources such as CertificateRequest seem like a natural fit for the cert-manager codebase. Better to have one source of truth for "how do I update a CR status condition".

Don't think it's a dealbreaker - but I wonder if we could add something to the cert-manager codebase instead? Doesn't really matter though